qualtran/bloqs/chemistry/prepare_mps/decompose_gate_hr_test.py

```diff
@@ -0,0 +1,145 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Tuple
+
+import numpy as np
+import pytest
+
+from qualtran import BloqBuilder
+from qualtran.bloqs.chemistry.prepare_mps.decompose_gate_hr import DecomposeGateViaHR
+from qualtran.bloqs.rotations.phase_gradient import PhaseGradientState
+from qualtran.testing import assert_valid_bloq_decomposition
+
+
+# these gates can be approximated exactly with the given phase_bitsize
+@pytest.mark.parametrize(
+    "phase_bitsize, gate_cols",
+    [
+        [2, ((0, ((0.5 + 0.5j), (0.5 - 0.5j))), (1, ((-0.5 - 0.5j), (0.5 - 0.5j))))],
+        [
+            4,
+            (
+                (
+                    0,
+                    (
+                        (-0.191341716182545 + 0.961939766255644j),
+                        (-0.038060233744357 + 0.191341716182545j),
+                    ),
+                ),
+                (
+                    1,
+                    (
+                        (0.038060233744356 - 0.191341716182545j),
+                        (-0.191341716182545 + 0.961939766255644j),
+                    ),
+                ),
+            ),
+        ],
+        [
+            2,
+            (
+                (0, ((-0 - 0.5j), (0.5 - 0j), (-0.5 + 0.5j), -0j)),
+                (1, ((0.5 + 0.5j), (0.5 - 0.5j), 0j, 0j)),
+                (2, (0.5j, (-0.5 + 0j), (-0.5 + 0.5j), 0j)),
+                (3, (0j, 0j, (-0 + 0j), (-1 + 0j))),
+            ),
+        ],
+    ],
+)
```

Comment:
super optional, but if you want the formatter to stop making this terrible, you can use

```python
# fmt: off
[ (2, (a, b)), ... ]
# fmt: on
```

and the formatter won't touch those lines